### ISSUE TITLE
Breaking change in 2.9.0?

### ISSUE THREAD

**Issue by DavidMStraub** [alisaifee/flask-limiter#381]
Hi,

I'm aware this is not a very useful bug report, sorry about that, but the unit tests for our project started failing today with the release of flask-limiter 2.9.0 and I have not found the root cause. Failed tests:
https://github.com/gramps-project/gramps-webapi/actions/runs/3780120902/jobs/6426053406

Pinning to 2.8.0 made the tests pass again.

Is there a breaking change we should be aware of?

Thanks for your work!

**Comment by alisaifee** [comment]
Hi David! My apologies, there was no intended breaking change in `2.9.0` (though I did have a tiny feeling...). I'll look into it and try to do a patch release asap.

**Comment by alisaifee** [comment]
Okay looks there's good news and bad news: 

## TL;DR:

- Good news: I don't think there's any real breaking change (though depending on whether this is a widespread issue, I might have to take that back).

- Bad news: Rate limiting afaict was not being applied in the gramps-api project (except when you upgraded to `2.9.0` - and then .. it was, which is why the tests started to fail). 

You could (probably) do one of:
1. Upgrade to `2.9.0` and disable rate limiting in the tests (`RATELIMIT_ENABLED=False` in the test config)
2. Fix the rate limit declaration according to [this recipe](https://flask-limiter.readthedocs.io/en/stable/recipes.html#using-flask-pluggable-views) in [token.py](https://github.com/gramps-project/gramps-webapi/blob/master/gramps_webapi/api/resources/token.py#L59), stay pinned to `2.8.x` and finally disable rate limiting in tests as per [1]. Once happy, remove the version constraint.

## The grampa story 
The problem is rooted in the way the rate limit is declared [here](https://github.com/alisaifee/gramps-webapi/blob/master/gramps_webapi/api/resources/token.py#L59) - i.e. decorating the method of a resource view. This was silently doing nothing in  < `2.9.0` since the `get/post/put...` methods of resource views are not actually view functions and those were the only functions that got rate limited when used with the `@limiter.limit` decorators. For rate limiting class based views [declaring the decorators on the `decorators` class attribute](https://flask-limiter.readthedocs.io/en/stable/recipes.html#using-flask-pluggable-views) had been the recommended approach (not just for flask-limiter, but generally if you want to apply a decorator to the eventual view function that is synthesized as described in the [flask docs on decorating class-based views](https://flask.palletsprojects.com/en/2.2.x/views/#view-decorators)).

As of `2.9.0` you can decorate any old function ([changelog reference](https://flask-limiter.readthedocs.io/en/stable/changelog.html#v2-9-0)) and when it is called from within a request context, it will deduct the rate limit from the endpoint the request was exercising. 

Since getting a token is in the setup of most of the request tests and the test app is shared across the test suite the request starts getting rejected due to the `1/second` limit.

**Comment by DavidMStraub** [comment]
Thanks a lot for the investigation and detailed explanation! Slightly embarrassing that it had no effect so far :blush: In hindsight it should have made us suspicious that the unit tests *don't* trigger it. I think your option 1 is the right way to fix it.

**Comment by alisaifee** [comment]
Please do go ahead and close the issue once you've verified that 2.9.0 does indeed work as originally expected! Good luck!

**Comment by alisaifee** [comment]
Looks like the issue is resolved in gramps-webapi so closing this.

**Comment by DavidMStraub** [comment]
Yes, sorry for not doing it earlier and thanks again for the help :rocket:
